cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginProperties.java

```diff
@@ -33,19 +35,40 @@
 
   // Currently only support String->String map.
   private final Map<String, String> properties;
+  private final Set<String> macros;
+
+  private PluginProperties(Map<String, String> properties, @Nullable Set<String> macros) {
+    this.properties = properties;
+    this.macros = macros;
+  }
 
   public static Builder builder() {
     return new Builder();
   }
 
   private PluginProperties(Map<String, String> properties) {
-    this.properties = properties;
+    this(properties, null);
   }
 
   public Map<String, String> getProperties() {
     return properties;
   }
 
+  @Nullable
+  public Set<String> getMacros() {
+    return macros;
+  }
+
+  /**
+   * Creates and returns a new instance of Plugin properties with current properties and the passed macros parameter.
+   * Note this is used by the Platform and its advicable for the user not to use this method.
```

Comment:
Can we suggest that they use something else instead? We have the Artifact RESTful API.
```
Note this is used internally by the CDAP Platform and it is advisable 
that end-users not use this method.